draft-packages/button/KaizenDraft/Button/styles.scss

```diff
@@ -0,0 +1,467 @@
+      calc(#{$ca-grid * 0.5} - #{$kz-border-focus-ring-border-width});
+  }
+
+  :global(.js-focus-visible) %caButton%caButtonForm:global(.focus-visible) & {
+    padding: calc(#{$ca-grid / 3} - #{$kz-border-focus-ring-border-width})
+      calc(#{$ca-grid * 1} - #{$kz-border-focus-ring-border-width});
+  }
+
+  :global(.js-focus-visible)
+    %caButtonSecondary%caButtonForm:global(.focus-visible)
+    & {
+    padding: calc(#{$ca-grid / 3} - #{$kz-border-focus-ring-border-width})
+      calc(#{$ca-grid * 0.5} - #{$kz-border-focus-ring-border-width});
+  }
+}
+
+%caButtonLabel {
+  // This line height equates to 3/4 of a grid unit - so 18px. We set this
+  // here rather than in ca-type-ideal-button because changing the type style
+  // changes the baseline grid alignment for all buttons. Changing it here keeps
+  // the baseline alignment correct for single line buttons.
+  line-height: 1;
+
+  &:only-child {
+    %caButtonSecondary & {
+      margin: 0;
+    }
+  }
+
+  &:not(:first-child) {
+    @include ca-margin($start: 0.5em);
+  }
+
+  &:not(:last-child) {
+    @include ca-margin($end: 0.5em);
+  }
+}
+
+%caButtonIconWrapper {
+  height: $caButtonIcon-height;
+  align-self: flex-start;
+}
+
+%caButtonIconButton {
+  width: $caButton-height;
+
+  // Reset for various states and variants
+  &:not(:disabled),
+  &:disabled {
+    &,
+    &:hover,
+    &:active,
+    &%caButtonPrimary,
+    &%caButtonSecondary {
+      color: $kz-color-wisteria-800;
+      background: none;
+      border-color: $kz-border-borderless-border-color;
+
+      &%caButtonReversed {
+        color: white;
+        &%caButtonPrimary {
+          %caButtonContent {
+            color: white;
+            background: none;
+          }
+        }
+      }
+    }
+  }
+  &%caButtonReversed,
+  :global(.js-focus-visible) &:global(.focus-visible) {
+    %caButtonContent {
+      padding: 0;
+    }
+  }
+
+  %caButtonContent {
+    justify-content: center;
+    width: 100%;
+    padding: 0;
+    // IE 10/11 workaround to fix vertical centering of icon
+    margin: calc(
+        (((#{$caButton-height}) - #{$caButtonIcon-height}) / 2) - #{$kz-border-solid-border-width}
+      )
+      0;
+  }
+
+  &%caButtonForm {
+    width: $caButton-formHeight;
+  }
+
+  &%caButtonDestructive:not(:disabled) {
+    &:hover,
+    &:active {
+      color: $kz-color-coral-600;
+    }
+
+    :global(.js-focus-visible) &:global(.focus-visible) {
+      border-color: $kz-color-cluny-500;
+    }
+  }
+}
+
+%caButtonDestructiveModifier {
+  &:not(:disabled) {
+    color: $kz-color-coral-600;
+
+    %caButtonLabel {
+      color: $kz-color-coral-600;
+    }
+  }
+
+  &:hover span,
+  &:active span {
+    color: $kz-color-coral-600;
+  }
+
+  &:hover,
+  &:active,
+  &:global(.focus-visible) {
+    // These are only used by consumers, so the !important is there to make sure these
+    // override any local competing styles that we can't foresee
```

Comment:
Should this have `:global(.js-focus-visible)` as well? - to follow the pattern of the other user `.focus-visible` usages (I guess it's trying to restrict that it's trying to restrict it so that it only applies if the plugin is actually used). I'm having a hard time understanding "These are only used by consumers" - is this something that should be deprecated? Is there anything special about this rule such that "to make sure these override any local competing styles that we can't foresee" only applies here and not everywhere else? I'm sure there's a reason, I'm just having a hard time following it.